File: tps/gabarito/tp01/TP01-Is-JAVA/Is.java

/**
 * @author: Thiago de Campos Ribeiro Nolasco
*/

import java.io.*;
class Is{

	/**
	 * receives a String and checks if it's composed only by integers
	 * @param String
	 * @return 
	*/
	public static boolean isFloat(String s){
		int count = 0, countChars = 0;
		for(int i = 0; i < s.length(); i++){
			if(s.charAt(i) == 44 || s.charAt(i) == 46) count++;
			if(s.charAt(i) >= 48 && s.charAt(i) <= 57) countChars++;
		}
//		if(count != 1 || countChars != s.length()) return false;
		if((count == 0 || count == 1) && countChars == s.length() - count) return true;
		return false;
	} // ending method



	public static boolean isInt(String s){
		for(int i = 0; i < s.length(); i++){
			if(s.charAt(i) < 48 || s.charAt(i) > 57) return false;
		}
		return true;
	} // ending method

	/**
	 * receives a String and checks if it's composed only by consonants
	 * @param String
	 * @return 
	*/
	public static boolean isConsoante(String s){
		char[] temp = new char[1];
		for(int i = 0; i < s.length(); i++){
			temp[0] = s.charAt(i);
			if((isVogal(temp)) || ((s.charAt(i) < 65 || s.charAt(i) > 90) && (s.charAt(i) < 97 || s.charAt(i) > 122))){
				return false;
			}
		}
		return true;
	} // ending method
	
	/**
	 * receives a array of char, transform it to String and sends it to isVogal (with diferent signature)
	 * @param char[]
	 * @return 
	*/ 
	public static boolean isVogal(char[] temp){
		String s = new String(temp);
		return isVogal(s);
	} // ending method

	/**
	 * receives a STRING and checks if it's composed only by vowels
	 * @param String
	 * @return 
	*/ 
	public static boolean isVogal(String s){
		for(int i = 0; i < s.length(); i++){
			if((s.charAt(i) != 65) && (s.charAt(i) != 69) && (s.charAt(i) != 73) && (s.charAt(i) != 79) && (s.charAt(i) != 85) &&
		           (s.charAt(i) != 97) && (s.charAt(i) != 101) && (s.charAt(i) != 105) && (s.charAt(i) != 111) && (s.charAt(i) != 117)){
			   	return false;
			   }	
		}
		return true;
	} // ending method

	/**
	 * receives a String and checks if it's equals to "FIM"
	 * @param String
	 * @return 
	*/
	public static boolean isFim(String s){
		return (s.length() == 3 && s.charAt(0) == 'F' && s.charAt(1) == 'I' && s.charAt(2) == 'M');
	} // ending method

	/**
	 * (MAIN) receives input from keyboard and starts the program
	 * @param: args String[]
	*/
	public static void main(String[] args) throws Exception{
		// data declaration
		BufferedReader leitor = new BufferedReader(new InputStreamReader(System.in));
		String[] entries = new String[1000];
		int numEntries = 0;
		String returned = "", result = ""; 

		// Reading entries
		do{
			entries[numEntries] = leitor.readLine();
		}while(!(isFim(entries[numEntries++])));
		numEntries--;

		for(int i = 0; i < numEntries; i++){
			result = "";

			returned = (isVogal(entries[i]))? "SIM" : "NAO";
			result += returned + " ";

			returned = (isConsoante(entries[i]))? "SIM" : "NAO";
			result += returned + " ";

			returned = (isInt(entries[i]))? "SIM" : "NAO";
			result += returned + " ";

			returned = (isFloat(entries[i]))? "SIM" : "NAO";
			result += returned;

			System.out.println(result);
		}

	}// ending main
		

}// ending class 

File: tps/gabarito/tp01/TP01-Is-C/Is.c
/**
 * @author: Thiago de Campos Ribeiro Nolasco
*/


// Biblioteca utilizada:
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#define true 1;
#define false 0;

// Prototipação:
int isFim(char s[]);
const char* isVogal(char s[]);
const char* isConsoante(char s[]);
const char* isInt(char s[]);
const char* isFloat(char s[]);

int main(void){
	
	// data declaration
	char str[1000], returned[3], result[16];
	int pos = 0;

	// CAN BE USED INSTED OF SCANF("%[^\n]"...):
	/*fgets(str, 1000, stdin);
	setbuf(stdin, NULL);
	if(str[strlen(str) - 1] == '\n'){
		str[strlen(str) - 1] = '\0';
	}*/
	
	scanf("%[^\n]", str);
	getchar();
	while(!(isFim(str))){
		

		printf("%s %s %s %s\n", isVogal(str), isConsoante(str), isInt(str), isFloat(str));
		scanf("%[^\n]", str);
		getchar();	
	}

	// Return without errors
	return 0;

}// ending main

// Checks if the received string is equals to "FIM"
int isFim(char s[]){
	return (strlen(s) == 3 && s[0] == 'F' && s[1] == 'I' && s[2] == 'M');
}// ending isFim

// Checks if the received string has only vowels
const char* isVogal(char s[]){
	for(int i = 0; i < strlen(s); i++){
		if((s[i] != 65) && (s[i] != 69) && (s[i] != 73) && (s[i] != 79) && (s[i] != 85) && (s[i] != 97) && (s[i] != 101) && (s[i] != 105) && (s[i] != 111) && (s[i] != 117)){
			return "NAO";		
		}
	}
	return "SIM";
}// ending isVogal

// Checks if the received string has 
const char* isConsoante(char s[]){
	for(int i = 0; i < strlen(s); i++){
		if((((s[i] == 65) || (s[i] == 69) || (s[i]) == 85) || (s[i] == 97) || (s[i] == 101) || (s[i] == 105) || (s[i] == 111) || (s[i] == 117)) || ((s[i] < 65 || s[i] > 90) && (s[i] < 97 || s[i] > 112))){
			return "NAO";
		}
	}
	return "SIM";
}// ending isConsoante

// Checks if the received string has only integers
const char* isInt(char s[]){
	for(int i = 0; i < strlen(s); i++){
		if(s[i] < 48 || s[i] > 57) return "NAO";
	}
	return "SIM";
}

const char* isFloat(char s[]){
	int count = 0, countChars = 0;
	for(int i = 0; i < strlen(s); i++){
		if(s[i] == 44 || s[i] == 46 || s[i] == 59) count++;
		if(s[i] >= 48 && s[i] <= 57) countChars++;
	}

	if((count == 0 || count == 1) && countChars == strlen(s) - count) return "SIM";
	return "NAO";
}
